### PR TITLE
refactor of f-header and f-footer country selector tests #trivial

### PR DESCRIPTION
**Pull request by LTurns**
### Changed

- Refactored large country selector test into two separate tests
- - Added string placeholder to tenant component tests

### CODE REVIEW

**Review by minrwhite — CHANGES_REQUESTED** [review]
I think we still need to have the link to DK/NO in the country selector, even though those countries are no longer on the JE platform?

oh sorry, my bad I thought they were causing a separate error 🤦 I'll re-add them now

File: packages/components/organisms/f-header/test/specs/component/desktop/f-header.component.spec.js [hunk 1]
```diff
@@ -62,8 +62,23 @@ describe('Desktop - f-header component tests', () => {
         });
     });
 
-    forEach([['gb', '.co.uk'], ['au', 'au'], ['at', 'at'], ['be', 'be-en'], ['bg', 'bg'], ['ca_en', 'skipthedishes.com'], ['ca_fr', 'skipthedishes.com/fr'], ['dk', '.dk'], ['jet_fr', '.fr'], ['de', '.de'], ['ie', '.ie'], ['il', '.il'], ['it', '.it'], 
-    ['lu', 'lu-en'], ['nl', '.nl'], ['nz', '.nz'], ['no', '.no'], ['pl', '.pl'], ['pt', '/pt'], ['ro', '/ro'], ['es', '.es'], ['ch_ch', '.ch'], ['ch_en', '/en'], ['ch_fr', '/fr'] ])
+    forEach([['gb', '.co.uk'], ['dk', '.dk'], ['no', '.no'], ['au', 'au'], ['at', 'at'], ['be', 'be-en'], ['bg', 'bg'], ['ca_en', 'skipthedishes.com'], ['jet_fr', '.fr'], ['de', '.de'], ['ie', '.ie'], ['il', '.il'], ['it', '.it']])
+    .it.only('should display all countries and redirect to correct URL', (expectedLocale, expectedUrl) => {
```
Comment (SachaZvetelman) [inline packages/components/organisms/f-header/test/specs/component/desktop/f-header.component.spec.js]: We probably don't want the `.only` here, right?

**Review by SachaZvetelman — DISMISSED** [review]

File: packages/components/organisms/f-footer/test/specs/component/shared/f-footer.component.spec.js [hunk 2]
```diff
@@ -106,8 +106,23 @@ describe('Shared - f-footer component tests', () => {
         expect(footer.isCurrentCountryIconDisplayed(expectedLocale)).toBe(true);
     });
 
-    forEach([['au', 'au'], ['at', 'at'], ['be', 'be-en'], ['bg', 'bg'], ['ca_en', 'skipthedishes.com'], ['ca_fr', 'skipthedishes.com/fr'], ['dk', '.dk'], ['jet_fr', '.fr'], ['de', '.de'], ['ie', '.ie'], ['il', '.il'], ['it', '.it'],
-        ['lu', 'lu-en'], ['nl', '.nl'], ['nz', '.nz'], ['no', '.no'], ['pl', '.pl'], ['pt', '/pt'], ['ro', '/ro'], ['es', '.es'], ['ch_ch', '.ch'], ['ch_en', '/en'], ['ch_fr', '/fr']])
+    forEach([['au', 'au'], ['at', 'at'], ['no', 'no'], ['dk', 'dk'], ['be', 'be-en'], ['bg', 'bg'], ['ca_en', 'skipthedishes.com'], ['jet_fr', '.fr'], ['de', '.de'], ['ie', '.ie'], ['il', '.il'], ['it', '.it']])
+    .it('should display all countries and redirect to correct URL', (country, expectedUrl) => {
```
Comment (minrwhite) [inline packages/components/organisms/f-footer/test/specs/component/shared/f-footer.component.spec.js]: I know this is the message signature as it existed previously, but might be good to use the [variable placeholders](https://www.npmjs.com/package/mocha-each#string-title) to insert the test case being run, to make failing cases easier to identify. Additionally, I for one would find it easier to parse if each case is on an individual line. e.g:

```suggestion
    forEach([
      ['au', 'au'],
      ['at', 'at'],
      ['no', 'no'],
      ['dk', 'dk'],
      ['be', 'be-en'],
      ['bg', 'bg'],
      ['ca_en', 'skipthedishes.com'],
      ['jet_fr', '.fr'],
      ['de', '.de'],
      ['ie', '.ie'],
      ['il', '.il'],
      ['it', '.it']
    ]).it('should display link for country code "%s" and redirect to correct URL', (country, expectedUrl) => { 
```

File: packages/components/organisms/f-footer/test/specs/component/shared/f-footer.component.spec.js [hunk 3]
```diff
@@ -106,8 +106,23 @@ describe('Shared - f-footer component tests', () => {
...
+        expect(footer.isCountryLinkItemDisplayed()).toBe(true);
+
+        // Act
+        footer.clickCountryLinkItem();
+
+        // Assert
+        expect(browser.getUrl()).toContain(expectedUrl);
+    });
+
+    forEach([['lu', 'lu-en'], ['nl', '.nl'], ['nz', '.nz'], ['pl', '.pl'], ['pt', '/pt'], ['ro', '/ro'], ['es', '.es'], ['ch_ch', '.ch'], ['ch_en', '/en'], ['ch_fr', '/fr']])
```
Comment (minrwhite) [inline packages/components/organisms/f-footer/test/specs/component/shared/f-footer.component.spec.js]: And here.

File: packages/components/organisms/f-header/test/specs/component/desktop/f-header.component.spec.js [hunk 4]
```diff
@@ -62,8 +62,23 @@ describe('Desktop - f-header component tests', () => {
         });
     });
 
-    forEach([['gb', '.co.uk'], ['au', 'au'], ['at', 'at'], ['be', 'be-en'], ['bg', 'bg'], ['ca_en', 'skipthedishes.com'], ['ca_fr', 'skipthedishes.com/fr'], ['dk', '.dk'], ['jet_fr', '.fr'], ['de', '.de'], ['ie', '.ie'], ['il', '.il'], ['it', '.it'], 
-    ['lu', 'lu-en'], ['nl', '.nl'], ['nz', '.nz'], ['no', '.no'], ['pl', '.pl'], ['pt', '/pt'], ['ro', '/ro'], ['es', '.es'], ['ch_ch', '.ch'], ['ch_en', '/en'], ['ch_fr', '/fr'] ])
+    forEach([['gb', '.co.uk'], ['dk', '.dk'], ['no', '.no'], ['au', 'au'], ['at', 'at'], ['be', 'be-en'], ['bg', 'bg'], ['ca_en', 'skipthedishes.com'], ['jet_fr', '.fr'], ['de', '.de'], ['ie', '.ie'], ['il', '.il'], ['it', '.it']])
+    .it('should display all countries and redirect to correct URL', (expectedLocale, expectedUrl) => {
```
Comment (minrwhite) [inline packages/components/organisms/f-header/test/specs/component/desktop/f-header.component.spec.js]: And here.

File: packages/components/organisms/f-header/test/specs/component/desktop/f-header.component.spec.js [hunk 5]
```diff
@@ -62,8 +62,23 @@ describe('Desktop - f-header component tests', () => {
...
+        expect(header.isCountryLinkDisplayed()).toBe(true);
+
+        // Act
+        header.clickCountryListItem();
+
+        // Assert
+        expect(browser.getUrl()).toContain(expectedUrl);
+    });
+
+    forEach([['lu', 'lu-en'], ['nl', '.nl'], ['nz', '.nz'], ['pl', '.pl'], ['pt', '/pt'], ['ro', '/ro'], ['es', '.es'], ['ch_ch', '.ch'], ['ch_en', '/en'], ['ch_fr', '/fr']])
```
Comment (minrwhite) [inline packages/components/organisms/f-header/test/specs/component/desktop/f-header.component.spec.js]: And here.

File: packages/components/organisms/f-footer/test/specs/component/shared/f-footer.component.spec.js (hunk 2 above)
Comment (LTurns) [reply packages/components/organisms/f-footer/test/specs/component/shared/f-footer.component.spec.js]: ooh this is awesome, thank you! We were discussing this yesterday in team web testers actually - I think some of these lengthy forEach's will end up going, but this is great to implement in the meantime and for the smaller iterable tests! 👍

**Review by minrwhite — APPROVED** [review]

**Review by SachaZvetelman — APPROVED** [review]

**Review by simonsandersje — APPROVED** [review]

**Pull request merged**